Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs

```diff
@@ -100,6 +100,18 @@ public async Task<IList<ConnectedEntity>> GetConnectedOrganisations(int organisa
         return await result.ToListAsync();
     }
 
+    public async Task<Organisation.LegalForm?> GetLegalForm(int organisationId)
+    {
+        var organisation = await context.Organisations
+            .Where(x => x.Id == organisationId && x.SupplierInfo != null)
+            .Include(x => x.SupplierInfo)
+                .ThenInclude(x => x.LegalForm)
```

Comment:
The PR is looking good but there are couple of dereference of a possibly null reference.

Reply:
This warning made no sense, but satisfied now.